spec/services/users/reset_progress_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+
+RSpec.describe Users::ResetProgress do
+  let(:user) { create(:user) }
+  let!(:foundations_path) { create(:path, default_path: true) }
+  let!(:foundations_course) { create(:course, path: foundations_path) }
```

Comment:
do we need to create a course and section? I thought the lesson factory might have created these for us. I could be misremembering that though.

Reply:
You're right. We do not need to create a course and section here. Although I don't see where/how the lesson factory creates them for us. I just used this pattern from the system tests without thinking too much about it. I think system tests need to create them, which I believe is so certain html elements are named and can be found by capybara.